soundcard/pulseaudio.py

```diff
@@ -86,6 +86,33 @@ def _block_operation(self, operation):
         while self._pa_operation_get_state(operation) == _pa.PA_OPERATION_RUNNING:
             time.sleep(0.001)
 
+    @property
+    def name(self):
+        """Return application name stored in client proplist"""
+        idx = self._pa_context_get_index(self.context)
+        # TODO: idx can be PA_INVALID_INDEX now
```

Comment:
How are constants like this accessible from Python if at all?

I think we should raise an exception here if it fails to retrieve the index.

Reply:
Usually, such constants are defined in enums, which are accessible as `_pa.CONSTANT_NAME` (look at `_pa.PA_SAMPLE_FLOAT32LE` for an example).

Reply:
Seems to be not the case, this is probably a preprocessor macro instead …